### ISSUE TITLE
[HTTP/3] Client closing control stream before closing connection

### ISSUE THREAD

**Issue by JamesNK** [dotnet/runtime#57021]
Justification for 6.0: Violation of HTTP/3 spec -- leading to error on the server side (which can be in theory ignored, but will be confusing for people adopting and experimenting with HTTP/3, lowering their confidence).

HttpClient is closing its outbound control stream before closing the connection. This is causing an `H3_CLOSED_CRITICAL_STREAM` error to be raised in Kestrel. There will be a race between that connection error, and `H3_NO_ERROR` error from the client closing the connection.

Spec says this stream must not be closed - https://quicwg.org/base-drafts/draft-ietf-quic-http.html#section-6.2.1-2. I believe the right behavior is to leave it open and allow it to be aborted with `H3_NO_ERROR` when the connection closes.

Kestrel logs:
```
    | [0.385s] Interop.FunctionalTests.Http3.Http3RequestTests Information: Disposing client.
    | [0.386s] Interop.FunctionalTests.Http3.Http3RequestTests Information: Waiting for server to receive connection close.
    | [0.396s] Microsoft.AspNetCore.Server.Kestrel.Http3 Debug: Connection id "0HMAPBN35QRHQ": HTTP/3 connection error.
    | Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.Http3ConnectionErrorException: HTTP/3 connection error (H3_CLOSED_CRITICAL_STREAM): The client closed the connection control stream inbound to the server.
    |    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.Http3ControlStream.HandleControlStream() in C:\Development\Source\AspNetCore\src\Servers\Kestrel\Core\src\Internal\Http3\Http3ControlStream.cs:line 238
    |    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.Http3ControlStream.ProcessRequestAsync[TContext](IHttpApplication`1 application) in C:\Development\Source\AspNetCore\src\Servers\Kestrel\Core\src\Internal\Http3\Http3ControlStream.cs:line 171
    | [0.398s] Microsoft.AspNetCore.Server.Kestrel.Transport.Quic Debug: Connection id "0HMAPBN35QRHQ" aborted by peer with error code 256.
    | System.Net.Quic.QuicConnectionAbortedException: Connection aborted by peer (256).
    |    at System.Net.Quic.Implementations.MsQuic.MsQuicConnection.AcceptStreamAsync(CancellationToken cancellationToken)
    |    at System.Net.Quic.QuicConnection.AcceptStreamAsync(CancellationToken cancellationToken)
    |    at Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.Internal.QuicConnectionContext.AcceptAsync(CancellationToken cancellationToken) in C:\Development\Source\AspNetCore\src\Servers\Kestrel\Transport.Quic\src\Internal\QuicConnectionContext.cs:line 96
```

**Comment by ManickaP** [comment]
@geoffkizer haven't you discussed this with quic work group? I remember that you pointed out it's weird to not to close the control stream before closing the connection, but I don't remember the outcome of that conversion.

Nevertheless, if this is the current state of specs, we should be following it.

**Comment by geoffkizer** [comment]
@jamesnk is correct, we should not close the control stream. We should just close the connection.

Note we still need to *dispose* the control stream, so that the msquic handle gets released. But we should do this after closing the connection.

**Comment by ManickaP** [comment]
Triage: this should be fixed for 6.0, temporarily assigning myself to not to punt it accidentally. Feel free to pick it up.
